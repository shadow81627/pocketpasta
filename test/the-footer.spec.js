import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import Component from '@/components/Layout/the-footer';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(Vuetify);

describe('Footer', () => {
  const factory = () =>
    shallowMount(Component, {
      localVue,
      data: () => ({
        version: '0.11.19',
        commit: 'a95a529a01ab4fe627cc276b35ed8f5e0c6f45d9',
      }),
    });

  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
