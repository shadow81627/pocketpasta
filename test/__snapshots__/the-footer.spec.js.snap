// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders properly 1`] = `
"<v-footer-stub height=\\"auto\\" tag=\\"div\\" tile=\\"true\\" class=\\"hidden-print-only\\"><v-card-stub loaderheight=\\"4\\" tag=\\"div\\" color=\\"transparent\\" flat=\\"true\\"><v-card-text-stub class=\\"text-muted\\"><span>Version: 0.11.19</span> <span> a95a529a01ab4fe627cc276b35ed8f5e0c6f45d9</span></v-card-text-stub></v-card-stub> <v-spacer-stub></v-spacer-stub> <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" color=\\"transparent\\" flat=\\"true\\"><v-card-text-stub class=\\"text-muted\\">This site is protected by reCAPTCHA and the Google
      <a href=\\"https://policies.google.com/privacy\\">Privacy Policy</a> and
      <a href=\\"https://policies.google.com/terms\\">Terms of Service</a> apply.
    </v-card-text-stub></v-card-stub></v-footer-stub>"
`;
